packages/@aws-cdk/toolkit/lib/toolkit/toolkit.ts

```diff
@@ -423,9 +423,16 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
           await ioHost.notify(info(`${chalk.cyan(stack.id)}.${chalk.cyan(name)} = ${chalk.underline(chalk.cyan(value))}`));
         }
 
-        await ioHost.notify(info('Stack ARN:'));
-
-        await ioHost.notify(data(deployResult.stackArn));
+        const obscuredTemplate = obscureTemplate(stack.template);
+        await ioHost.notify(info(`Stack ARN:${deployResult.stackArn}`, 'CDK_TOOLKIT_I0002', {
+          stack: {
+            stackName: stack.stackName,
+            hierarchicalId: stack.hierarchicalId,
+            template: stack.template,
+            stringifiedJson: serializeStructure(obscuredTemplate, true),
+            stringifiedYaml: serializeStructure(obscuredTemplate, false),
+          },
```

Comment:
i am not quite sure exactly what information we want to send via this channel but i mostly copied what was done on synth